src/tabs/tab-header-bar.scss

```diff
@@ -74,28 +75,75 @@ $label-horizontal-spacing: awsui.$space-l;
   flex-shrink: 0;
   display: flex;
   max-inline-size: calc(90% - #{$label-horizontal-spacing});
-
-  & > button {
-    background-color: transparent;
-  }
 }
 
 .tabs-tab-label {
   display: flex;
   align-items: center;
-  padding-block: awsui.$space-scaled-2x-xxs;
   padding-inline: $label-horizontal-spacing;
   text-align: start;
   position: relative;
 
   @include styles.text-wrapping;
 }
 
+.tabs-tab-header-container {
+  position: relative;
+
+  border-block: awsui.$border-divider-section-width solid transparent;
+  border-inline: awsui.$border-divider-section-width solid transparent;
+
+  padding-block: awsui.$space-scaled-2x-xxs;
+  padding-inline: $label-horizontal-spacing;
+
+  &,
+  & > button {
+    background-color: transparent;
+    display: flex;
+    align-items: center;
+    text-align: start;
+  }
+}
+
+.tabs-tab-header-container:not(.tabs-tab-disabled) {
+  // This is the underline for the currently selected tab.
+  &:after {
+    content: '';
+    position: absolute;
+    inset-inline-start: 0;
+    inline-size: 100%;
+    inset-block-end: calc(-1 * #{awsui.$border-divider-section-width});
+    block-size: awsui.$border-active-width;
+    border-start-start-radius: awsui.$border-radius-tabs-focus-ring;
+    border-start-end-radius: awsui.$border-radius-tabs-focus-ring;
+    border-end-start-radius: awsui.$border-radius-tabs-focus-ring;
+    border-end-end-radius: awsui.$border-radius-tabs-focus-ring;
+    background: awsui.$color-border-tabs-underline;
+    opacity: 0;
+  }
+
+  &.refresh {
+    &:after {
+      @include styles.with-motion {
+        transition: opacity awsui.$motion-duration-refresh-only-medium awsui.$motion-easing-refresh-only-c;
+      }
+    }
+  }
+}
+
 .tabs-tab:not(:last-child) {
-  & > a > .tabs-tab-label,
-  & > button > .tabs-tab-label {
+  & > .tabs-tab-header-container {
     margin-inline-end: calc(-1 * #{awsui.$border-divider-section-width});
-    border-inline-end: awsui.$border-divider-section-width solid $separator-color;
+    &:before {
+      content: '';
+      block-size: 1.7rem;
```

Comment:
I think this could be a regression: previously if a tab label spanned multiple lines, the divider would be the full height, but now it would be fixed to always `1.7rem`. I'm not sure if this is something we actively want to support or not though. @timogasda @pan-kot can we check with design?

Reply:
Good callout, I feel like they'd lean towards having a dynamic height - I believe I originally did this because the height would be the full height of the new container div which seemed too tall